### ISSUE TITLE
Unstable version won't install

### ISSUE THREAD

**Issue by esamo** [chocolatey/ChocolateyGUI#203]
Hey.

I've tried to install `Skype` via `ChocolateyGUI v0.12.0-unstable012`, but the action gets canceled after displaying 3/4 of the status window covered in red text that I can not read in 200 miliseconds.

Are those errors logged somewhere?

---

Regards & respect.

**Comment by gep13** [comment]
@esamo ChocolateyGUI is simply a wrapper around Chocolatey, so if there is a problem installing a package you should be able to look at the Chocolatey logs to see what is going on.

These logs can be found here:

```c:\ProgramData\chocolatey\chocolateyinstall```

**NOTE:** this assumes a fresh install on Chocolatey 0.9.8.28.  If you are using an earlier version, you can find the above in ```C:\chocolatey\chocolateyinstall```

**Comment by esamo** [comment]
`Skype` install done fine by command version of `chocolatey`.

---

Found the logs in `C:\ProgramData\chocolatey\chocolateyinstall`. And a big one `chocolateyInstall.log`.

But I can't find phrase `skype` anywhere before command installation. Actually I can not find anything helpful. Only few entries of `list` requests. Like the problem wasn't logged here or at all.

---

Regards & respect.

**Comment by gep13** [comment]
@esamo I have just tried to reproduce the problem that you have described on a fresh VM.  This is what I have done:

* Install Chocolatey
* Install ChocolateyGUI
* Use ChocolateyGUI to install Skype

All of the above worked fine, without any problems.

If you can provide a set of repeatable steps to show this problem in action, I can dig into it further, but for now, I don't think there is anything else I can add.  Sorry!

**Comment by esamo** [comment]
Even update doesn't work for me on unstable version. I've found the line containing `exited with '0'`:

`20150113-19:05:44 [DEBUG] Command ["C:\ProgramData\chocolatey\chocolateyinstall\nuget.exe" list -Prerelease -NonInteractive -Source "C:\ProgramData\chocolatey\lib"] exited with '0'.`

And the log below

---

20150113-19:05:36 [DEBUG] Running 'Append-InstallLog' with chocoInstallLog:''
20150113-19:05:37 [CHOCO] ################################################################################
20150113-19:05:37 [CHOCO] #                  Going Chocolatey on 2015-01-13 19:05:37Z                    #
20150113-19:05:37 [CHOCO] ################################################################################
20150113-19:05:37 [DEBUG] Arguments: $command = 'list'|$force=False
|$prerelease=False|$packageNames=''
|$source=''
|$version=''|$allVersions=False
|$overrideArguments=False|$InstallArguments=''
|$localonly=True|$verbosity=False|$debug=False|$quiet=False
|$name=''|$ignoreDependencies=False|$forceX86=False
|$packageParameters=''
|PowerShellVersion=2.0|OSVersion=6.1.7601.65536
20150113-19:05:37 [DEBUG] Current Security Protocol - Ssl3, Tls
20150113-19:05:37 [DEBUG] Running 'Chocolatey-List' with selector: '', source:''
20150113-19:05:37 [DEBUG] Showing prerelease versions of packages
20150113-19:05:37 [DEBUG] Executing command ["C:\ProgramData\chocolatey\chocolateyinstall\nuget.exe" list -Prerelease -NonInteractive -Source "C:\ProgramData\chocolatey\lib"]
20150113-19:05:44 [DEBUG] Command ["C:\ProgramData\chocolatey\chocolateyinstall\nuget.exe" list -Prerelease -NonInteractive -Source "C:\ProgramData\chocolatey\lib"] exited with '0'.
20150113-19:05:44 [DEBUG] Running 'Get-ConfigValue' with configValue:'ksMessage'
20150113-19:05:44 [DEBUG] Running 'Get-UserConfigValue' with configValue:'ksMessage'
20150113-19:05:44 [DEBUG] After checking the user config the value of 'ksMessage' is ''
20150113-19:05:44 [DEBUG] Value not found in the user config file - checking the global config
20150113-19:05:44 [DEBUG] Running 'Get-GlobalConfigValue' with configValue:'ksMessage'
20150113-19:05:44 [DEBUG] After checking the global config the value of 'ksMessage' is 'true'

---

Regards & respect

**Comment by RichiCoder1** [comment]
Call me crazy, but why is this there:
' -Source "C:\ProgramData\chocolatey\lib"'?

**Comment by gep13** [comment]
> Isn't that the switch to tell it where it should be retrieving packages from?

Yes, it is.  This shouldn't be required, unless you have all your source chocolatey packages downloaded into that folder, which really isn't a good idea.

**Comment by esamo** [comment]
This is how GUI configured itself. I don't know. Command version of `choco` doesn't fill `source` param.

Regards & respect.

**Comment by RichiCoder1** [comment]
Just realized that's because that's the log for a 'clist -lo' command. That shouldn't have any relevance to the issue.

**Comment by gep13** [comment]
So exactly what command are you executing?

I am not clear on whether you are using Chocolatey from the command line, or ChocolateyGUI.

**Comment by esamo** [comment]
I tried to install / uninstall / update via GUI, but while it doesn't work for me, I'm using command-line version. Reporting the problem with GUI.

Logs from `C:\Users\esamo\AppData\Roaming\ChocolateyGUI\Logs` .

---

This one on `uninstall` trial:

2015-01-14 19:23:53,559 [1] ERROR ChocolateyGui.Services.ChocolateyService [(null)] - ExecutePackageCommmand threw an exception.
System.Management.Automation.PSSecurityException: File C:\ProgramData\chocolatey\chocolateyinstall\chocolatey.ps1 cannot be loaded because the execution of scripts is disabled on this system. Please see "get-help about_signing" for more details. ---> System.UnauthorizedAccessException: File C:\ProgramData\chocolatey\chocolateyinstall\chocolatey.ps1 cannot be loaded because the execution of scripts is disabled on this system. Please see "get-help about_signing" for more details.
   --- Koniec śladu stosu wyjątków wewnętrznych ---
   w System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input)
   w System.Management.Automation.Runspaces.Pipeline.Invoke()
   w System.Threading.Tasks.Task`1.InnerInvoke()
   w System.Threading.Tasks.Task.Execute()
--- Koniec śladu stosu z poprzedniej lokalizacji, w której wystąpił wyjątek ---
   w System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   w System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   w System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   w ChocolateyGui.Services.ChocolateyService.<RunDirectChocolateyCommand>d__5a.MoveNext()
--- Koniec śladu stosu z poprzedniej lokalizacji, w której wystąpił wyjątek ---
   w System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   w System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   w ChocolateyGui.Services.ChocolateyService.<ExecutePackageCommand>d__53.MoveNext()

---

Older ones

2015-01-02 15:31:25.9435 1 [FATAL] Chocolatey.Explorer.Startup Fatal exception: StructureMap Exception Code:  207
Internal exception while creating Instance 'b347d2c0-af74-4d27-bf8d-ab691ff06434' of PluginType Chocolatey.Explorer.Powershell.IRunAsync.  Check the inner exception for more details.

2015-01-02 15:31:25.7277 1 [DEBUG] Chocolatey.Explorer.Services.SourceService.SourceService Initialize
2015-01-02 15:31:25.7589 1 [DEBUG] Chocolatey.Explorer.Services.SourceService.SourceService Loadsources
2015-01-02 15:31:25.7745 1 [DEBUG] Chocolatey.Explorer.Services.FileStorageService.LocalFileSystemStorageService Load document: sources.xml
2015-01-02 15:31:25.8031 1 [DEBUG] Chocolatey.Explorer.Services.SourceService.SourceService Added source: Chocolatey.org http://chocolatey.org/api/v2 True
2015-01-02 15:31:25.8187 1 [DEBUG] Chocolatey.Explorer.Services.SourceService.SourceService Added source: Nuget https://packages.nuget.org/v1/FeedService.svc True
2015-01-02 15:31:25.8187 1 [DEBUG] Chocolatey.Explorer.Services.SourceService.SourceService Sources changed: System.Collections.Generic.List`1[Chocolatey.Explorer.Model.Source]
2015-01-02 15:31:25.8343 1 [DEBUG] Chocolatey.Explorer.Services.SourceService.SourceService Current source: Chocolatey.org (http://chocolatey.org/api/v2)
2015-01-02 15:31:25.8343 1 [DEBUG] Chocolatey.Explorer.Services.SourceService.SourceService Current source changed: Chocolatey.org (http://chocolatey.org/api/v2)
2015-01-02 15:31:25.8655 1 [DEBUG] Chocolatey.Explorer.Services.PackagesService.CachedAvailablePackagesService Invalidate cache
2015-01-02 15:31:25.8811 1 [DEBUG] Chocolatey.Explorer.Services.PackageVersionService.CachedPackageVersionService Invalidate cache

---

Regards & respect

**Comment by gep13** [comment]
The first error message, looks like your system isn't configured to allow execution of scripts.  Has something changed recently on your system?

With regard to the "older ones" that error message is from an old version of ChocolateyGUI, which had a number of issues around dependenices, and having to manually run as an administrator.  I think we can ignore that error message.

What is ```Advancved SystemCare```?

**Comment by gkingston** [comment]
I get the same error message with any attempt to install.  The error window will pop up and disappear pretty quickly.  From the log file: `System.Management.Automation.PSSecurityException: File C:\ProgramData\chocolatey\chocolateyinstall\chocolatey.ps1 cannot be loaded because running scripts is disabled on this system.`

This is on a pretty fresh Windows 8.1 install from 3 weeks ago.  I haven't messed with any security settings.  Running `choco install <packagename>` from an admin powershell prompt works fine(that's how I installed this preview among other packages).  I can't install anything with the preview GUI version.  Using Chocolatey 0.9.8.32 and ChocolateyGUI 0.12.0-unstable0127

**Comment by gep13** [comment]
@gkingston can you run the following command:

```
Get-ExecutionPolicy
```

And let me know what it returns.

**Comment by gkingston** [comment]
Just one line:  Restricted.  This is running powershell as admin.  Like I said above, I can install stuff with the command line version of chocolatey fine.

**Comment by gep13** [comment]
When you install from Chocolatey Command Line, are you using a PowerShell Window, or a Command Prompt Window?  We probably need to handle this better for this particular scenario, but with the pending release of Chocolatey vNext which is based on C# rather than PowerShell, and the subsequent re-write of ChocolateyGUI to use that, I think this problem will go away.

**Comment by gkingston** [comment]
Powershell as admin is how I have always ran the command line chocolatey.

**Comment by gep13** [comment]
> Powershell as admin is how I have always ran the command line chocolatey.

Hmm, I am really confused by what could be going on here.  I am going to have to take some time to look into this.

**Comment by esamo** [comment]
I'm using the command prompt window though, having same situation as far as I understand.

Regards & respect

**Comment by esamo** [comment]
The problem persists even after full reinstall of `choco` and its' `GUI`.
While trying to install a any package, status box gets covered in red text that disappears too fast to read.

Regards & respect.

**Comment by gep13** [comment]
Esamo,

Can you please confirm what version of Chocolatey you are using, and also what version of ChocolateyGUI?

Also, can you grab a screenshot of the error that you are seeing?

Thanks

Gary

Sent from my Windows Phone
________________________________

**Comment by esamo** [comment]
ChocolateyGUI version `0.12.3`
Chocolatey version `0.9.8.33`

Eventually I managed to capture the error and adjust contrast to read. There it is:

![](http://oi59.tinypic.com/2f09jtc.jpg)

1. I have to research that on my own or You guys know what how to fix that and if will this mess the security?

2. Honestly no offence, but what is the point of displaying error for a nano second?

3. Also I found out that GUI searches for `packages` way faster (maybe up to 10x) than the `command prompt`, while GUI is not working for me, it would be nice to know how to speed up `choco search` results. Does it ask a server for results or search through downloaded database?

---

Regards & respect
